### ISSUE TITLE
allow SpatialDropout for Conv layers

### ISSUE THREAD

**Issue by bertsky** [Calamari-OCR/calamari#351]
The only Dropout layer available uses keras.layers.Dropout:

https://github.com/Calamari-OCR/calamari/blob/2f71b7eb08339d25ccb21d80c1d5b851f3d5bdaa/calamari_ocr/ocr/model/layers/dropout.py#L27

However, it seems customary to use [SpatialDropout1D](https://keras.io/api/layers/regularization_layers/spatial_dropout1d/) after the convolutional layers.
